torch/utils/tensorboard/beholder/beholder.py

```diff
@@ -0,0 +1,225 @@
+        elif config['values'] == 'arrays':
+            # print('===arrays===')
+            final_image = np.concatenate([arr for arr, _ in arrays])
+            stat = self.stats(arrays)
+            write_pickle(
+                stat, '{}/{}'.format(self.PLUGIN_LOGDIR, SECTION_INFO_FILENAME))
+        elif config['values'] == 'trainable_variables':
+            # print('===trainable===')
+            final_image = np.concatenate([arr for arr, _ in trainable])
+            stat = self.stats(trainable)
+            write_pickle(
+                stat, '{}/{}'.format(self.PLUGIN_LOGDIR, SECTION_INFO_FILENAME))
+        if len(final_image.shape) == 2:  # Map grayscale images to 3D tensors.
+            final_image = np.expand_dims(final_image, -1)
+
+        return final_image
+
+    def _enough_time_has_passed(self, FPS):
+        '''For limiting how often frames are computed.'''
+        if FPS == 0:
+            return False
+        else:
+            earliest_time = self.last_update_time + (1.0 / FPS)
+            return time.time() >= earliest_time
+
+    def _update_frame(self, trainable, arrays, frame, config):
+        final_image = self._get_final_image(config, trainable, arrays, frame)
+        self._write_summary(final_image)
+        self.last_image_shape = final_image.shape
+
+        return final_image
+
+    def _update_recording(self, frame, config):
+        '''Adds a frame to the current video output.'''
+        # pylint: disable=redefined-variable-type
+        should_record = config['is_recording']
+
+        if should_record:
+            if not self.is_recording:
+                self.is_recording = True
+                print('Starting recording using %s',
+                      self.video_writer.current_output().name())
+            self.video_writer.write_frame(frame)
+        elif self.is_recording:
+            self.is_recording = False
+            self.video_writer.finish()
+            print('Finished recording')
+
+    # TODO: blanket try and except for production? I don't someone's script to die
+    #       after weeks of running because of a visualization.
+    def update(self, trainable=None, arrays=None, frame=None):
+        '''Creates a frame and writes it to disk.
+
+        Args:
+            trainable: a list of namedtuple (tensors, name).
+            arrays: a list of namedtuple (tensors, name).
+            frame: lalala
+        '''
+
+        new_config = self._get_config()
+        if True or self._enough_time_has_passed(self.previous_config['FPS']):
+            # self.visualizer.update(new_config)
+            self.last_update_time = time.time()
+            final_image = self._update_frame(
+                trainable, arrays, frame, new_config)
+            self._update_recording(final_image, new_config)
+
+    ##############################################################################
+    # @staticmethod
+    # def gradient_helper(optimizer, loss, var_list=None):
```

Comment:
Is this stuff we're going to bring back? Feel free to uncomment if that's the case so I know what to review. Thanks!